tests/libvmi/network.go

```diff
@@ -48,3 +53,28 @@ func InterfaceDeviceWithMasqueradeBinding() kvirtv1.Interface {
 		},
 	}
 }
+
+// ValidateVMIandPodIPMatch Checks that the vmi pod and vmi scheme have matching Ip/Ips fields for primary interface
+func ValidateVMIandPodIPMatch(vmi *v1.VirtualMachineInstance, vmiPod *k8sv1.Pod) error {
+	if vmi.Status.Interfaces[0].IP != vmiPod.Status.PodIP {
+		return fmt.Errorf("Status.Interfaces[0].IP %s does not equal pod's Status.PodIP %s",
+			vmi.Status.Interfaces[0].IP, vmiPod.Status.PodIP)
+	}
+
+	if len(vmi.Status.Interfaces[0].IPs) != len(vmiPod.Status.PodIPs) {
+		return fmt.Errorf("Status.Interfaces[0].IPs len does not equal pod's Status.PodIPs len")
+	}
+
+	if len(vmi.Status.Interfaces[0].IPs) == 0 {
+		return fmt.Errorf("Status.Interfaces[0].IPs len is zero")
+	}
+
+	for i, ip := range vmiPod.Status.PodIPs {
```

Comment:
This will not work in case the array and the map have different order
you should search by same index for both VMI IP and pod IP 
https://github.com/kubevirt/kubevirt/pull/4067/commits/cf8e94e6139ae8396f2adeb84ee58b61bf020016#diff-d47bf036797965d742615e64adc3b777R2530

Reply:
the size is validated couple lines above it, i dont need by the same range, as we spoke (Miguel and me and you)

Reply:
I explained on that conversation why you should, do what you think

Reply:
so explain again please, because one of us is wrong / doesnt understand

Reply:
ping (edited)

Reply:
> This will not work in case the array and the map have different order
> you should search by same index for both VMI IP and pod IP

@ormergi , both seem to be slices, could you please explain which map you refer to here?